src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1024,17 +1017,11 @@ def coarse_block_replacement(self):
 
         """
         self.debug('Zeroing blocks')
-        i = 0
-        while i < len(self.last_data.blocks):
+        for u, v in self.last_data.blocks:
```

Comment:
I’d like @DRMacIver to double-check this, because we’ve discussed this in the past, and I’m pretty sure there’s a good reason for writing the loop this way. I think it’s to do with the case where the underlying list mutates during the iteration, but I can’t remember the exact details at 5am on a Monday.

Unlike the other examples of cleanup, which are mostly just whitespace or similar cleanup, I’d be *super* careful about editing this block. I‘d want to understand why it was originally written this way before changing it.

Reply:
No, re-reading that more carefully you're right!  I'll add some comments explaining it instead to avoid this coming up again.